src/com/akshay/captain/jira.main.kts

```diff
@@ -191,39 +282,159 @@ fun <E> List<E>.ifNotEmpty(function: () -> Unit): List<E> {
+        .get()
+        .build()
+
 
     try {
         client.newCall(request).enqueue(object : Callback {
             override fun onFailure(call: Call, e: IOException) {
             }
 
             override fun onResponse(call: Call, response: Response) {
+                handleAllPreviousReleases(response.body,releaseNotes,appVersion,versionCode,captain)
+
+            }
+        })
+
+    } catch (e: IOException) {
+        e.printStackTrace()
+    }
+}
+fun handleAllPreviousReleases(
+    responseBody: ResponseBody?,
+    releaseNotes: String,
+    appVersion: String,
+    versionCode: String,
+    captain: String
+) {
+    val paraEnd = "</p>"
+    val res = responseBody?.string()
+    val output = JsonParser().parse(res).asJsonObject
+    val confluenceVersion = output.get("version").asJsonObject.get("number").asString.toInt()
+    val title = output.get("title").asString
+    val body = output.get("body").asJsonObject.get("view").asJsonObject.get("value")
+
+    val data = body.toString().split(paraEnd)
+    val newPostFixData = data.takeLast(data.count().minus(1)).joinToString(separator = paraEnd)
+    val generatedData = generateReleaseNote(
+        releaseNotes,
+        appVersion,
+        versionCode,
+        captain,
+    )
+
+    val newData = data.first().plus(paraEnd).plus("<p /><p />").plus(generatedData).plus(newPostFixData)
+    updateReleaseNotes(
+        title,
+        confluenceVersion,
+        newData,
+        appVersion
+    )
+}
+/**
+ * Appends the new release notes to the existing releases
+ */
+fun updateReleaseNotes(title: String, version: Int, release: String, appVersion: String) {
+    val client = OkHttpClient().newBuilder().addInterceptor(BasicAuthInterceptor()).build()
+
+    val payload = JSONObject()
+    try {
+        val versionObject = JSONObject().also {
+            it["number"] = version.plus(1)
+        }
+        val bodyObject = JSONObject().also { storage ->
+            storage["storage"] = JSONObject().also { body ->
+                body["value"] = release
+                body["representation"] = "storage"
+            }
+        }
+        payload["version"] = versionObject
+        payload["type"] = "page"
+        payload["title"] = title
+        payload["body"] = bodyObject
+    } catch (e: JSONException) {
+        e.printStackTrace()
+    }
+
+
+    val payloadString = payload.toString()
+    val payloadStringUnEscaped = StringEscapeUtils.unescapeJava(payloadString).toString()
+    val prefixPayloadStringUnescaped = payloadStringUnEscaped.replace("\"\"<p>", "\"<p>").replace("</p>\"\"", "</p>\"")
+    val newPayloadStringUnescaped = prefixPayloadStringUnescaped.replace("</p>\"\"", "</p>\"")
+    val mediaType = "application/json; charset=utf-8".toMediaType()
+    val body = newPayloadStringUnescaped.toRequestBody(mediaType)
+    val request: Request = Request.Builder()
+        .url("${clueJiraUrl}/wiki/rest/api/content/3490545669")
```

Comment:
suggestion: have this constant template on top for easy edition

Reply:
I cannot do it, as soon as I move it to top of file, it starts compilation error!

Reply:
compilation error? but I mean, we could have "$s//wiki/rest/api/content/3490545669" and use string.format here. But is just a suggestion if we want all the strings in one place instead of hardocded here, but leave it here, is not critical nor needed n.n